nomos-da-core/kzg/src/lib.rs

```diff
@@ -0,0 +1,89 @@
+use kzg::eip_4844::{
+    blob_to_kzg_commitment_rust, bytes_to_blob, compute_blob_kzg_proof_rust,
+    verify_blob_kzg_proof_rust,
+};
+use kzg::types::fr::FsFr;
+use kzg::types::g1::FsG1;
+use kzg::types::kzg_settings::FsKZGSettings;
+use kzg_traits::G1;
+use std::error::Error;
+
+// TODO: Can this be imported?
+pub const FIELD_ELEMENTS_PER_BLOB: usize = 4096;
+pub const BYTES_PER_BLOB: usize = BYTES_PER_FIELD_ELEMENT * FIELD_ELEMENTS_PER_BLOB;
+pub const BYTES_PER_FIELD_ELEMENT: usize = 32;
+pub const BYTES_PER_PROOF: usize = 48;
+pub const BYTES_PER_COMMITMENT: usize = 48;
+
+pub struct Commitment(pub(crate) FsG1);
+
+pub struct Proof(pub(crate) FsG1);
+
+impl Commitment {
+    pub fn as_bytes_owned(&self) -> [u8; BYTES_PER_COMMITMENT] {
+        self.0.to_bytes()
+    }
+}
+
+impl Proof {
+    pub fn as_bytes_owned(&self) -> [u8; BYTES_PER_PROOF] {
+        self.0.to_bytes()
+    }
+}
+
+pub fn compute_commitments(
+    data: &[u8],
+    settings: &FsKZGSettings,
+) -> Result<Vec<Commitment>, Box<dyn Error>> {
+    let blobs = data.chunks(BYTES_PER_BLOB).map(bytes_to_blob);
+    let mut res = Vec::new();
+    for blob in blobs {
+        let mut blob = blob?;
+        let blob_len = blob.len();
+        // FIXME: is this actually ok?
+        // fill the blob with zeros if it is not full
+        if blob.len() != BYTES_PER_BLOB {
+            let offset = BYTES_PER_BLOB - blob_len;
+            blob.extend((0..offset).map(|_| FsFr::default()));
+        }
```

Comment:
I guess this is ok as long as we always pad/unpad data whenever we do KZG computations.

Reply:
This changes in another PR that will be merged on top of this one (#325 )